File: Zig/src/k04/k04.zig
const std = @import("std");
const golem = @import("../structs/golem.zig");
const utility = @import("../utility/utility.zig");

fn question_1() anyerror!void {
    const stdin = std.io.getStdIn();

    try utility.prompt("年齢を入力してください。＞");
    var buffer: [100]u8 = undefined;
    var input = (try utility.readNextLine(stdin.reader(), &buffer)).?;
    const age = try std.fmt.parseInt(i32, input, 10);

    if (age < 3 or age >= 70) {
        std.log.info("入場料金無料です。", .{});
    } else {
        std.log.info("通常料金です。", .{});
    }
}

fn question_2() anyerror!void {
    const stdin = std.io.getStdIn();

    try utility.prompt("性別を選択してください。（０：男性　１：女性）＞");
    var buffer: [100]u8 = undefined;
    var input = (try utility.readNextLine(stdin.reader(), &buffer)).?;
    const gender = try std.fmt.parseInt(i32, input, 10);

    switch (gender) {
        0 => std.log.info("あら、格好良いですね。", .{}),
        1 => std.log.info("あら、モデルさんみたいですね。", .{}),
        else => std.log.info("そんな選択肢はありません。", .{})
    }
}

fn question_3() anyerror!void {
    const stdin = std.io.getStdIn();

    try utility.prompt("年齢を入力してください。＞");
    var buffer: [100]u8 = undefined;
    var input = (try utility.readNextLine(stdin.reader(), &buffer)).?;
    const age = try std.fmt.parseInt(i32, input, 10);

    if (age < 3 or age >= 70) {
        std.log.info("入場料金無料です。", .{});
    } else if (age >= 3 and age <= 15) {
        std.log.info("子供料金で半額です。", .{});
    } else if (age >= 60 and age < 70) {
        std.log.info("シニア割引で一割引きです。", .{});
    } else {
        std.log.info("通常料金です。", .{});
    }
}

fn question_4() anyerror!void {
    std.log.info("＊＊＊おみくじプログラム＊＊＊", .{});

    const stdin = std.io.getStdIn();

    try utility.prompt("おみくじを引きますか　（はい：１　いいえ：０）＞");
    var buffer: [100]u8 = undefined;
    var input = (try utility.readNextLine(stdin.reader(), &buffer)).?;
    const choice = try std.fmt.parseInt(i32, input, 10);

    if (choice >= 1) {
        var prng = std.rand.DefaultPrng.init(blk: {
            var seed: u64 = undefined;
            try std.os.getrandom(std.mem.asBytes(&seed));
            break :blk seed;
        });
        const rand = &prng.random;

        switch (rand.intRangeAtMost(i32, 0, 4)) {
            0 => std.log.info("大吉　とってもいいことがありそう！！", .{}),
            1 => std.log.info("中吉　きっといいことあるんじゃないかな", .{}),
            2 => std.log.info("小吉　少しぐらいはいいことあるかもね", .{}),
            3 => std.log.info("凶　今日はおとなしくておいた方がいいかも", .{}),
            4 => std.log.info("大凶　これじゃやばくない？早く家に帰った方がいいかも", .{}),
            else => {}
        }
    }
}

pub fn execute(num: usize) anyerror!void {
    switch (num) {
        1 => try question_1(),
        2 => try question_2(),
        3 => try question_3(),
        4 => try question_4(),
        else => {},
    }
}

File: Erlang/hello_world/src/questions/k04.erl
-module(k04).

-export([execute/1]).

-spec question_1() -> 'ok'.
question_1() ->
    {ok, [Age]} = io:fread("年齢を入力してください。＞", "~d"),
    io:format(if Age < 3 orelse Age >= 70 -> "入場料金無料です。~n";
                 true -> "通常料金です。~n"
    end).

-spec question_2() -> 'ok'.
question_2() ->
    {ok, [Gender]} = io:fread("性別を選択してください。（０：男性　１：女性）＞", "~d"),
    io:format(case Gender of
                0 -> "あら、格好良いですね。~n";
                1 -> "あら、モデルさんみたいですね。~n";
                _ -> "そんな選択肢はありません。~n"
    end).

-spec question_3() -> 'ok'.
question_3() ->
    {ok, [Age]} = io:fread("年齢を入力してください。＞", "~d"),
    io:format(case Age of
                X when X < 3 orelse X >= 70 -> "入場料金無料です。~n";
                X when X >= 3, X =< 15 -> "子供料金で半額です。~n";
                X when X >= 60, X < 70 -> "シニア割引で一割引きです。";
                _ -> "通常料金です。~n"
    end).

-spec question_4() -> 'ok'.
question_4() ->
    io:format("＊＊＊おみくじプログラム＊＊＊~n"),
    {ok, [Choice]} = io:fread("おみくじを引きますか　（はい：１　いいえ：０）＞", "~d"),
    if Choice >= 1 ->
        Oracle = rand:uniform(5) - 1,
        io:format(case Oracle of
                    0 -> "大吉　とってもいいことがありそう！！~n";
                    1 -> "中吉　きっといいことあるんじゃないかな~n";
                    2 -> "小吉　少しぐらいはいいことあるかもね~n";
                    3 -> "凶　今日はおとなしくておいた方がいいかも~n";
                    4 -> "大凶　これじゃやばくない？早く家に帰った方がいいかも~n";
                    _ -> ""
        end);
        true -> io:format("")
    end.

-spec execute(_) -> 'ok'.
execute(1) -> question_1();
execute(2) -> question_2();
execute(3) -> question_3();
execute(4) -> question_4();
execute(_) -> 'ok'.